core/io/file_access_zip.cpp

```diff
@@ -45,7 +45,9 @@ static void *godot_open(void *data, const char *p_fname, int mode) {
 	}
 
 	FileAccess *f = (FileAccess *)data;
-	f->open(p_fname, FileAccess::READ);
+	if (!f->is_open()) {
```

Comment:
What happens if a file is already open? Won't it fail opening the new file as requested?

Reply:
It's not clear if it wants to open a different file.   We might want to compare the file paths to make sure.

The integration of FileAccess with the uzip lib isn't super clean.  It seems like uzip doesn't want to start with an open file, just enough information to do it on its own.  But we give it a `FileAccess::open` anyway.  uzip then called the open callback `godot_open` to open the file, even though we gave it to uzip already opened. `godot_open` then calls `FileAccess::open`  again and seems to leave it dangling.  :grimacing: 

We might want to consider starting uzip with `FileAccess::create` and then call reopen unguarded.